app/DTO/AlbumProtectionPolicy.php

```diff
@@ -2,50 +2,66 @@
 
 namespace App\DTO;
 
-class AlbumProtectionPolicy extends DTO
-{
-	public const IS_PUBLIC_ATTRIBUTE = 'is_public';
-	public const REQUIRES_LINK_ATTRIBUTE = 'requires_link';
-	public const IS_NSFW_ATTRIBUTE = 'is_nsfw';
-	public const IS_DOWNLOADABLE_ATTRIBUTE = 'is_downloadable';
-	public const IS_SHARE_BUTTON_VISIBLE_ATTRIBUTE = 'is_share_button_visible';
-	public const GRANTS_FULL_PHOTO_ATTRIBUTE = 'grants_full_photo';
-
-	public bool $isPublic;
-	public bool $requiresLink;
-	public bool $isNSFW;
-	public bool $isDownloadable;
-	public bool $isShareButtonVisible;
-	public bool $grantsFullPhoto;
+use App\Contracts\AbstractAlbum;
+use App\Models\Extensions\BaseAlbum;
+use App\SmartAlbums\BaseSmartAlbum;
 
+/**
+ * This defines the Protection policy.
+ *
+ * In other words this gives the info of the security attributes of an album.
+ * It should not be used in the front end to determine whether an action is doable or not (e.g. share link available).
+ */
+class AlbumProtectionPolicy extends ArrayableDTO
+{
 	public function __construct(
-		bool $isPublic,
-		bool $requiresLink,
-		bool $isNSFW,
-		bool $isDownloadable,
-		bool $isShareButtonVisible,
-		bool $grantsFullPhoto
+		public bool $is_public,
+		public bool $is_link_required,
+		public bool $is_nsfw,
+		public bool $is_share_button_visible,
+		public bool $grant_access_full_photo,
+		public bool $grant_download,
```

Comment:
```suggestion
		public bool $grants_full_photo_access,
		public bool $grants_download,
```

3rd-person-singular-s, please. I won't blue-pencil every occurrence of this during this review, because this is something which is much more easily corrected consistently in a proper IDE which supports refactoring.